### PR TITLE
Revert 133b70c0 for emacs < 25

### PR DESCRIPTION
**Pull request by Kha**
There just doesn't seem to be a way to do this with emacs < 25: http://emacs.stackexchange.com/a/12456

@leodemoura Does this fix it?

### CODE REVIEW

File: src/emacs/lean-server.el [hunk 1]
```diff
@@ -42,17 +42,29 @@
...
+                 :coding 'utf-8
+                 :noquery t
+                 :sentinel #'lean-server-handle-signal
+                 :stderr (lean-server-stderr-buffer-name))
+                (progn (start-file-process "lean-server"
+                                    (format "*lean-server (%s)*" (buffer-name))
+                                    (lean-get-executable lean-executable-name)
+                                    "--server"
+                                    (format "*%s*" (buffer-name)))
+                       (set-process-coding-system lean-server-process 'utf-8 'utf-8)
```
Comment (gebner) [inline src/emacs/lean-server.el]: Is `lean-server-process` already set here?

**Comment by leodemoura** [comment]
> @leodemoura Does this fix it?

No, I'm getting the following error:

```
Error: (wrong-type-argument processp nil)
```

**Comment by Kha** [comment]
Sorry for that, this time I've tested both branches.

**Comment by leodemoura** [comment]
> Sorry for that, this time I've tested both branches.

It is working again :)

**Comment by leodemoura** [comment]
Sorry, false alarm. I got the following errors:

```
error in lean-server command handler: (json-readtable-error)
error in process filter: tq-process-buffer: Spurious communication from process lean-server<2>, see buffer *spurious*
error in process filter: Spurious communication from process lean-server<2>, see buffer *spurious*
error in process filter: tq-process-buffer: Spurious communication from process lean-server<2>, see buffer *spurious*<2>
error in process filter: Spurious communication from process lean-server<2>, see buffer *spurious*<2>
error in process filter: tq-process-buffer: Spurious communication from process lean-server<2>, see buffer *spurious*<3>
error in process filter: Spurious communication from process lean-server<2>, see buffer *spurious*<3>
Mark set
End of buffer [2 times]
End of buffer
```

**Comment by leodemoura** [comment]
Emacs also opened 3 `*spurious*` buffers

**Comment by leodemoura** [comment]
I'm using Emacs 24.5.1, and tried to open `tests/lean/run/converter.lean`

**Comment by Kha** [comment]
Oh dear, _this_ one was from https://github.com/leanprover/lean/commit/133b70c0e34d04dee5efb3dca09b211b2e91db16#diff-fd5f593186df1a749a43347112fac7f8L228, which turns out not to actually be necessary. I've reverted that too now.

Just to be clear, assertion errors under emacs < 25 will still create the `Spurious communication` errors, while they go to a separate buffer under emacs >= 25.

**Comment by leodemoura** [comment]
Thanks.
It seems to be working now.

**Pull request closed without merging**